policy/predicate/author.go

```diff
@@ -29,17 +30,30 @@ type HasAuthorIn struct {
 
 var _ Predicate = &HasAuthorIn{}
 
-func (pred *HasAuthorIn) Evaluate(ctx context.Context, prctx pull.Context) (bool, string, error) {
+func (pred *HasAuthorIn) Evaluate(ctx context.Context, prctx pull.Context) (bool, *common.PredicateInfo, error) {
 	author := prctx.Author()
 
 	result, err := pred.IsActor(ctx, prctx, author)
 	desc := ""
-
+	contributorInfo := common.ContributorInfo{
+		RequiredMembership: common.Actors{
+			Organizations: pred.Organizations,
+			Teams:         pred.Teams,
+			Users:         pred.Users,
+		},
+		ActualAuthor: author,
+	}
+	predicateInfo := common.PredicateInfo{
+		Type:            common.ContributorType,
+		Name:            "HasAuthorIn",
+		ContributorInfo: &contributorInfo,
+	}
```

Comment:
(4/?) To give an example of how the proposed `PredicateInfo` struct might work in a predicate:

```go
info := PredicateInfo{
  ValuesPhrase: "authors",
  Values: []string{author},
  ConditionPhrase: "meet the required membership conditions",
  ConditionsMap: map[string][]string{
    "Organizations": pred.Organizations,
    "Teams": pred.Teams,
    "Users": pred.Users,
  },
}
```